superset/config.py

```diff
@@ -364,6 +364,24 @@ class CeleryConfig(object):
 # Interval between consecutive polls when using Hive Engine
 HIVE_POLL_INTERVAL = 5
 
+
+# System to handle delegated data access. Implement both is_allowed_access() and
+# is_eligible_datasource() to delegate access controls.
+class PermsDecider:
```

Comment:
Shouldn't that be a static class (not meant to be instantiated)? Also note that in py2 it's always preferable to derive `object`  as in `class PermsDecider(object):`.

Also this base class should probably be declared outside of `config.py` (maybe security.py)?

What about simplifying the interface with a simple `has_datasource_access(user, datasource)` callable that only applies if it's defined?

Reply:
Agree that it should be a static class. 
The concern is that if it is defined outside of config.py, there's no way to delegate access decisions to a system outside of superset. 

Since perms_decider might only apply to a subset of databases ot datasources, If we use has_datasource_access, it's not clear what to return when the datasource in question is out of the jurisdiction of the perms_decider.

Reply:
The interface could be such that `has_datasource_access` could return `None` to say that it's out of its jurisdiction.

```python
def has_datasource_access(user, datasource):
    if in_my_jurisdiction(datasource):
        return has_access_according_to_me(user, datasource)
    return None
```

Reply:
But None is falsey and so it will be easy to mix up `access denied` from `I don't know enough to decide for this case`